packages/jellyfish-api-core/__tests__/category/account/futureswap.test.ts

```diff
@@ -0,0 +1,744 @@
+        { tokenAmount: '1@DFI', currency: 'USD' },
+        { tokenAmount: '10000@BTC', currency: 'USD' },
+        { tokenAmount: '2@TSLA', currency: 'USD' },
+        { tokenAmount: '4@AMZN', currency: 'USD' },
+        { tokenAmount: '1@DUSD', currency: 'USD' }
+      ]
+    }
+  )
+  await testing.generate(1)
+
+  // collateral tokens
+  await testing.rpc.loan.setCollateralToken({
+    token: 'DFI',
+    factor: new BigNumber(1),
+    fixedIntervalPriceId: 'DFI/USD'
+  })
+  await testing.generate(1)
+
+  await testing.rpc.loan.setCollateralToken({
+    token: 'BTC',
+    factor: new BigNumber(0.5),
+    fixedIntervalPriceId: 'BTC/USD'
+  })
+  await testing.generate(1)
+
+  // loan token
+  await testing.rpc.loan.setLoanToken({
+    symbol: 'TSLA',
+    fixedIntervalPriceId: 'TSLA/USD'
+  })
+  await testing.generate(1)
+
+  await testing.rpc.loan.setLoanToken({
+    symbol: 'AMZN',
+    fixedIntervalPriceId: 'AMZN/USD'
+  })
+  await testing.generate(1)
+
+  await testing.rpc.loan.setLoanToken({
+    symbol: 'DUSD',
+    fixedIntervalPriceId: 'DUSD/USD'
+  })
+  await testing.generate(1)
+
+  idTSLA = await testing.token.getTokenId('TSLA')
+  idDUSD = await testing.token.getTokenId('DUSD')
+
+  // create a vault and take loans
+  const vaultAddr = await testing.generateAddress()
+  const vaultId = await testing.rpc.loan.createVault({
+    ownerAddress: vaultAddr,
+    loanSchemeId: 'default'
+  })
+  await testing.generate(1)
+
+  await testing.rpc.loan.depositToVault({
+    vaultId: vaultId, from: collateralAddress, amount: '100000@DFI'
+  })
+
+  // wait till the price valid.
+  await testing.container.waitForPriceValid('TSLA/USD')
+
+  // take multiple loans
+  await testing.rpc.loan.takeLoan({
+    vaultId: vaultId,
+    to: collateralAddress,
+    amounts: ['300@TSLA', '500@DUSD', '100@AMZN']
+  })
+  await testing.generate(1)
+
+  // Futures setup
+  // set the dfip2203/active to false
+  await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/active': 'false' } })
+  await testing.generate(1)
+
+  // set dfip2203 params
+  futInterval = 25
+  futRewardPercentage = 0.05
+  await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/reward_pct': futRewardPercentage.toString(), 'v0/params/dfip2203/block_period': futInterval.toString() } })
+  await testing.generate(1)
+
+  // activat the dfip2203/active now
+  await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/active': 'true' } })
+  await testing.generate(1)
+
+  // Retrive and verify gov vars
+  const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+  expect(attributes.ATTRIBUTES['v0/params/dfip2203/active']).toStrictEqual('true')
+  expect(attributes.ATTRIBUTES['v0/params/dfip2203/reward_pct']).toStrictEqual(futRewardPercentage.toString())
+  expect(attributes.ATTRIBUTES['v0/params/dfip2203/block_period']).toStrictEqual(futInterval.toString())
+}
+
+async function getNextSettleBlock (): Promise<number> {
```

Comment:
I use this function too in my PR. Do you want to create this as a global function in masternode.ts?

Reply:
I also use this function in my PR.

Reply:
use `await testing.container.call('getfutureswapblock',[])`. simple math, we can rely on ain returned value.